src/Api/Commands/MergeDiscussionHandler.php

```diff
@@ -123,6 +140,41 @@ public function handle(MergeDiscussion $command)
                     $this->catchError($e, 'updating');
                 }
 
+                // Users who were following the merged discussions should now follow the new one.
+                if ($this->extensions->isEnabled('flarum-subscriptions')) {
+                    // Move subscriptions to the new discussion.
+                    $this->connection->table('discussion_user')
+                        ->whereIn('discussion_id', $discussions->pluck('id'))
+                        ->where('subscription', 'follow')
+                        ->whereNotExists(function (Builder $query) use ($discussion) {
+                            $query->selectRaw(1)
+                                ->from('discussion_user', 'discussion_user2')
+                                ->where('discussion_user2.discussion_id', $discussion->id)
+                                ->whereColumn('discussion_user.user_id', 'discussion_user2.user_id');
+                        })
+                        ->update([
+                            'discussion_id' => $discussion->id,
+                            'last_read_post_number' => $discussion->last_post_number,
```

Comment:
Does this mean the new merged discussion will be marked completely as read if any of the merged ones was being followed by the user? Even if the user hadn't read any of the posts involved in the merge?

Reply:
Yea, it only applies of they've never read the target discussion though, setting the number to 1 might be better in retrospect.

Reply:
I think i'd rather set it to the last post number where the user has read all previous posts - but that requires either checking old & new post numbers or created times. More complicated.

Reply:
I'm not sure that it's doable, post numbers change when merging from discussion A, B, C to discussion Y. Which post number to pick from A B C and how to pick it with its new value seems like a lot of complexity.

Also, since they haven't read discussion Y at all, setting a post number other than 1 means original posts they haven't read gets marked as read.

Reply:
Yeah, so what I described was keeping the original "last read post" references of each discussion and comparing their new post numbers, picking the smallest one as the one to save in the new discussion. This way it'd resume them at the point where new posts appear that they may have not read, with the ones they definitely had before them.

Eg.
| Original Discussion ID | Last Read Post's Old Number | Last Read Post's New Number|
|--------|--------|--------|
| 1 | 4 | 6 |
| 2 | 1 | 3 | 
| 3 | 1 | 8 | 

We know they must've read posts #1-3 at least in the merged discussion, so we can set last post read number to 3. This doesn't necessarily mean the next post will not have been read, but it guarantees the ones beforehand were.

Of course, this could be also done by calculating which post is first encountered in the new numbering that wasn't read, but that seems more consuming. 

Or we can do none of this cause It's not worth it 🤷. Not in the scope of this PR anyways.
I did start second guessing myself while I wrote this 😅 but I think what I wrote is correct.

Reply:
Hmm, thinking back again maybe it's too much to introduce this PR logic in the extension, might be better to keep it simple. How about just a `MergingDiscussions` event dispatched instead. That way it can be used for custom behavior without burdening current extension.